Automattic/woocommerce-payments
# feat: express payment methods with grouped settings

Fixes https://github.com/Automattic/woocommerce-payments/issues/1905

#### Changes proposed in this Pull Request









Splitting up a bigger PR.
This ensures that the settings that are saved to the backend are honored by the frontent.

There's already a separate task that ensures that the settings are migrated: https://github.com/Automattic/woocommerce-payments/issues/1991

#### Testing instructions





- Go to http://localhost:8082/wp-admin/admin.php?page=wc-settings&tab=checkout&section=woocommerce_payments_digital_wallets
- Change some settings
- Save
- Ensure you're _not_ using localhost (use Jurassic Tube or ngrok instead)
- Ensure you have some cards saved on GPay or on Chrome autofill or on ApplePay - see requirements here: https://stripe.com/docs/js/payment_request/can_make_payment
- Settings are honored on the frontend

PLEASE NOTE: I noticed that changing the settings often might not reflect the changes immediately to the frontend, probably due to some caching. If the changes aren't reflected, just wait a little bit. Example: https://d.pr/i/t8b93E

Examples:

<details>
<summary>"Donate" - Large - Dark (GPay)</summary>

![Screen Shot 2021-06-03 at 5 36 06 PM](https://user-images.githubusercontent.com/273592/120720734-347feb80-c492-11eb-9d3a-d50929e7ed5d.png)

</details>


<details>
<summary>"Only icon" - Default - Dark (GPay)</summary>

![Screen Shot 2021-06-03 at 5 14 13 PM](https://user-images.githubusercontent.com/273592/120720398-9f7cf280-c491-11eb-8315-4c5d22a75a04.png)

</details>

<details>
<summary>"Buy" - Default - Dark (GPay)</summary>

![Screen Shot 2021-06-03 at 5 00 20 PM](https://user-images.githubusercontent.com/273592/120720454-b885a380-c491-11eb-9bb5-49c8d878527b.png)

</details>

<details>
<summary>"Buy" - Default - Dark (Autofill)</summary>

![Screen Shot 2021-06-03 at 4 53 32 PM](https://user-images.githubusercontent.com/273592/120720526-d9e68f80-c491-11eb-96ef-9e2f80a01275.png)

</details>



-------------------
